lib/spack/spack/patch.py

```diff
@@ -99,16 +104,20 @@ def apply(self, stage):
             stage: stage for the package that needs to be patched
         """
         patch = which("patch", required=True)
+        args = ['-s', '-p', str(self.level), '-i', self.path, "-d",
+                self.working_dir]
+        if self.ignore_whitespace:
+            args += ['--ignore-whitespace']
         with working_dir(stage.source_path):
             # Use -N to allow the same patches to be applied multiple times.
```

Comment:
Possibly a digression, but this comment struck me as odd because the `-N` argument isn't actually used when invoking `patch` below. It wouldn't be hard to add this feature also.